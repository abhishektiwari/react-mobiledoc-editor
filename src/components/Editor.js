import PropTypes from 'prop-types';
import React from 'react';

class Editor extends React.Component {
  constructor(props) {
    super(props);
    this.editor = React.createRef();
  }

  componentDidMount() {
    const { editor } = this.context;
    if (editor) {
      editor.render(this.editor.current);
    }
  }

  render() {
    const props = { ...this.props, ref: this.editor };
    return <div {...props} />;
  }
}

Editor.contextTypes = {
  editor: PropTypes.object
};

export default Editor;
