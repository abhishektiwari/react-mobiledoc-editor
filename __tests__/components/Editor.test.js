import React from 'react';
import { shallow, mount } from 'enzyme';
import Editor from '../../src/components/Editor';

describe(Editor, () => {
  it('mounts Mobiledoc.Editor', () => {
    const context = { editor: { render: jest.fn() }};
    mount(<Editor />, { context });

    expect(context.editor.render).toBeCalled();
  });

  it('passes arbitrary props to node', () => {
    const wrapper = shallow(<Editor className="foo" />);
    expect(wrapper).toMatchSnapshot();
  });
});
