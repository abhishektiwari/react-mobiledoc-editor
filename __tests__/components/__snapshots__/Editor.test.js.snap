// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes arbitrary props to node 1`] = `
<div
  className="foo"
/>
`;
